src/controller/CHIPDevice.cpp

```diff
@@ -292,6 +292,16 @@ CHIP_ERROR Device::OpenPairingWindow(uint32_t timeout, PairingWindowOption optio
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR Device::UpdateAddress(const Transport::PeerAddress & addr)
+{
+    bool didLoad;
+    ReturnErrorOnFailure(LoadSecureSessionParametersIfNeeded(didLoad));
+
+    mDeviceUdpAddress = addr;
```

Comment:
please check that addr is of `Type::kUdp` since that seems to be an invariant the code tries to enforce.

Reply:
I can do it, but I assume eventually we will go in the opposite direction and rename `mDeviceUdpAddress` to `mDeviceAddress` to allow for TCP connectivity as well, right?